api/migrations/20200626194403-removeOrphanedDocs.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+let dbm;
+let type;
+let seed;
+
+let ObjectID = require('mongodb').ObjectID;
+
+/**
+  * We receive the dbmigrate dependency from dbmigrate initially.
+  * This enables us to not have to rely on NODE_PATH.
+  */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  try {
+    let mClient = await db.connection.connect(db.connectionString, { native_parser: true });
+    let nrptiCollection = mClient.collection('nrpti');
+    console.log('Getting all records');
+    const records = await nrptiCollection.find(
+      {
+        _schemaName: {
+          $in: [
+            'AdministrativePenalty',
+            'AdministrativeSanction',
+            'Agreement',
+            'Certificate',
+            'ConstructionPlan',
+            'CourtConviction',
+            'Inspection',
+            'ManagementPlan',
+            'Order',
+            'Permit',
+            'RestorativeJustice',
+            'SelfReport',
+            'Ticket',
+            'Warning'
+          ]
+        }
+      }
+    ).toArray();
+    let recordDocIds = [];
+    for (let i = 0; i < records.length; i++) {
+      if (records[i].documents && records[i].documents.length > 0) {
```

Comment:
You could probably put this in the `.find` and make mongodb do the work.

```
    const records = await nrptiCollection.find(
      {
        _schemaName: {
          $in: [...]
        },
        $where: 'this.documents.length > 0'
      },
    )
    .project({ documents: 1 })
    .toArray();

    let recordDocs = records.map(rec => rec.documents);
    let recordDocIds = recordDocs.concat(...recordDocs);
```

Note: NEVER use `$where` in an aggregation on the API. Slow as molasses.